### ISSUE TITLE
How to handle Error sending message: [429] Too Many Requests?

### ISSUE THREAD

**Issue by gekoramy** [rubenlagus/TelegramBots#238]
My bot sends some messages with inline buttons.
I can easily simulate a DoS attack in less than 15 seconds: by simply tapping repeatedly those buttons.
How can I avoid this situation?

Thanks for the attention.

**Comment by Tar4s** [comment]
@gekoramy 
It's impossible. The blocking is triggered at the telegram servers side. But you can write own block :)

**Comment by Clevero** [comment]
@JetCoder1 if I understod your PR at #230 correctly, your code would "fix" that issue, doesn't it?

**Comment by JetCoder1** [comment]
**@Clevero**, yes, for most cases.

**Comment by rubenlagus** [comment]
@gekoramy it is already merged, can you check it and close the issue if done?

**Comment by gekoramy** [comment]
Unfortunatelly it doesn't solve the problem.
I found out that there's also a max number (should be 20) of messages sendable to the same chat per minute.
I adapted the @JetCoder1 solution with a limit of 10 messages (to the same chat) per minute.
However with just one person tapping repeatedly (and rapidly) inline buttons it still throws me:
Error sending message: [429] Too Many Requests.

I really don't understand why Telegram doesn't share the actual limits.
